app/views/shared/_post_layout.html.erb

```diff
@@ -0,0 +1,8 @@
+<div class="post-section">
+  <a href='<%= request.path %>/<%=post.id%>'><p>Post #<%= index + 1 %></p></a>
+  <p class="post-text"><%=post.text%> </p>
+  <div class="reaction">
+    <p> Likes: <%= Like.where('post_id = ?', post.id).count %></p>
+    <p> Comments:<%= Comment.where('post_id = ?', post.id).count %></p>
```

Comment:
- [ ] Here as well. Avoid using logic in the view instead use helper or model for that.

`Like.where('post_id = ?', post.id).count` and `Comment.where('post_id = ?', post.id).count` these line have logic please move this in helper method.

Reply:
Hi @Dipeshtwis, thank you for the review. However, I wanted to ask one thing. Is it possible to call helpers from within the view? Because otherwise if I call the helper in the controller, I will have to pass the array of likes and comments for each post.id and then loop through that array in the view.

Reply:
Got it. Found this [article[(https://www.rubyguides.com/2020/01/rails-helpers/)